src/guiFormSpecMenu.cpp

```diff
@@ -3311,24 +3311,30 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			(m_selected_item->listname == s.listname) &&
 			(m_selected_item->i == s.i);
 
-		// buttons: 0 = left, 1 = right, 2 = middle
-		// up/down: 0 = down (press), 1 = up (release), 2 = unknown event, -1 movement
+		// buttons: 0 = left, 1 = right, 2 = middle, 3 = wheeldown, 4 = wheelup
+		// up/down: 0 = down (press or wheel), 1 = up (release),
+		//  2 = unknown event, -1 movement
 		int button = 0;
 		int updown = 2;
-		if (event.MouseInput.Event == EMIE_LMOUSE_PRESSED_DOWN)
-			{ button = 0; updown = 0; }
-		else if (event.MouseInput.Event == EMIE_RMOUSE_PRESSED_DOWN)
-			{ button = 1; updown = 0; }
-		else if (event.MouseInput.Event == EMIE_MMOUSE_PRESSED_DOWN)
-			{ button = 2; updown = 0; }
-		else if (event.MouseInput.Event == EMIE_LMOUSE_LEFT_UP)
-			{ button = 0; updown = 1; }
-		else if (event.MouseInput.Event == EMIE_RMOUSE_LEFT_UP)
-			{ button = 1; updown = 1; }
-		else if (event.MouseInput.Event == EMIE_MMOUSE_LEFT_UP)
-			{ button = 2; updown = 1; }
-		else if (event.MouseInput.Event == EMIE_MOUSE_MOVED)
-			{ updown = -1;}
+		switch (event.MouseInput.Event) {
+		case EMIE_LMOUSE_PRESSED_DOWN:
+			button = 0; updown = 0; break;
+		case EMIE_RMOUSE_PRESSED_DOWN:
+			button = 1; updown = 0; break;
+		case EMIE_MMOUSE_PRESSED_DOWN:
+			button = 2; updown = 0; break;
+		case EMIE_MOUSE_WHEEL:
+			button = 3 + (event.MouseInput.Wheel > 0); updown = 0; break;
+		case EMIE_LMOUSE_LEFT_UP:
+			button = 0; updown = 1; break;
+		case EMIE_RMOUSE_LEFT_UP:
+			button = 1; updown = 1; break;
+		case EMIE_MMOUSE_LEFT_UP:
+			button = 2; updown = 1; break;
+		case EMIE_MOUSE_MOVED:
+			updown = -1;
+		default:;
```

Comment:
Works fine here: https://github.com/minetest/minetest/blob/master/src/chat.cpp#L619
Perhaps try adding a `break;` to the line above?